src/xdocs/property_types.xml

```diff
@@ -831,16 +831,62 @@
 
     <section name="SeverityLevel">
       <p>
-        This type represents the severity level of a check violation. The
-        valid options are:
+        This type represents the severity level of a module violation.
+        The following table describes the valid options:
       </p>
 
-      <ul>
-        <li><code>ignore</code></li>
-        <li><code>info</code></li>
-        <li><code>warning</code></li>
-        <li><code>error</code></li>
-      </ul>
+      <div class="wrapper">
+        <table>
+          <tr>
+            <td>Option</td>
+            <td>Definition</td>
+          </tr>
+
+          <tr>
+            <td><code>ignore</code></td>
+            <td>
+              If a violation occurs, modules with this severity are ignored as if they never
+              occurred.
+              Checkstyle will not display the violation and will not fail the build.
+              This severity can be used to disable a module in the configuration. However,
+              depending on the setting of <code>executeIgnoredModules</code> when executed, the
+              module may still run and if an exception occurs, the exception will turn into an
+              error and act as one.
+            </td>
+          </tr>
+
+          <tr>
+            <td><code>info</code></td>
+            <td>
+              If a violation occurs, modules with this severity are displayed as informational.
+              Checkstyle will display the violation, but will not fail the build.
+              This severity can be used for normal logging.
+            </td>
+          </tr>
+
+          <tr>
+            <td><code>warning</code></td>
+            <td>
+              This severity behaves exactly the same as <code>info</code> in Checkstyle.
+              It can be used to stress something that <b>MAY</b> cause future problems and may
+              need to be looked into.
+              If executing Checkstyle through Ant,
+              <a href="anttask.html#Parameters"><code>maxWarnings</code></a> can be used to pass
+              the build until a certain number of warnings are found.
+            </td>
+          </tr>
+
+          <tr>
+            <td><code>error</code></td>
+            <td>
+              If a violation occurs, modules with this severity are displayed as error and should
+              be considered as a failure.
```

Comment:
modules are not shown, we show violations. you might mean `violation with`.

>be considered as a failure.

Something tel me that we should avoid phases on "failure", as it is decision of plugin/cli/.... to make decision. Plugin can do failure based on "gates" , for example info > 200.

Reply:
> modules are not shown, we show violations. you might mean violation with.

Modules are the ones where severities are set in the config. I think I was trying to reference that.

How about we rewrite these sentences to:
* If a violation with this severity occurs, it is ...

> we should avoid phases on "failure"

I am avoiding using the same words as the name of the severity. Otherwise we are saying "Error is an error" and it means nothing for describing to the user if they were already confused about "error" in the first place.

I don't have anything else as I do think this is a violation that needs corrective action. Do we want to use that wording instead?